app/models/reservation.rb

```diff
@@ -120,6 +120,9 @@ def self.upcoming_offline(start_at_limit)
       .where(product_id: OfflineReservation.current.pluck(:product_id))
       .not_canceled
       .not_ended
+      .joins(:order_detail)
+      .where("order_details.state" => %w(new inprocess))
+      .where("order_details.problem" => false)
```

Comment:
Is it possible (outside changing the database directly) to have an `OrderDetail` be a problem order and also be in `new` or `inprocess`? Just wondering if it's necessary.